src/validation.cpp

```diff
@@ -3369,7 +3372,10 @@ bool Chainstate::InvalidateBlock(BlockValidationState& state, CBlockIndex* pinde
 
     // Only notify about a new block tip if the active chain was modified.
     if (pindex_was_in_chain) {
-        m_chainman.GetNotifications().blockTip(GetSynchronizationState(IsInitialBlockDownload()), *to_mark_failed->pprev);
+        // Ignoring return value for now, this could be changed to bubble up
+        // kernel::Interrupted value to the caller so the caller could
+        // distinguish between completed and interrupted operations.
```

Comment:
I think I was confusing the kernel/node boundaries a bit. You're right that here in the kernel logic that is `validation.cpp`, the documentation and behaviour are what we'd expect.

My concern was around the _node_ implementation of `blockTip()` now potentially calling `StartShutdown()` and returning a `kernel:Interrupted` without leaving any trace. It's not something we'd expect to happen in `InvalidateBlock()` (and it would be behaviour change). I think I was wrong in suggesting we catch/log/document that here in `validation.cpp` though, as it's node specific behaviour.

Perhaps it would be beneficial to move [this logging](https://github.com/bitcoin/bitcoin/commit/09938a41d904c05b4676b064da9baa85e53e3e6f#diff-114c2880ec1ff2c5293ac65ceda0637bf92c05745b74b58410585a549464a33fR937) to `KernelNotifications::blocksImported()` and add a similar logging statement in `KernelNotifications::blockTip`?

...
_Although having typed all of this out... Is `InvalidateBlock()` even meant to be/stay in kernel?_